controllers/cluster_plugins.go

```diff
@@ -54,5 +55,10 @@ func (r *ClusterReconciler) updatePluginsStatus(ctx context.Context, cluster *ap
 		cluster.Status.PluginStatus[i].BackupCapabilities = entry.BackupCapabilities
 	}
 
+	// If nothing changes, there's no need to hit the API server
+	if reflect.DeepEqual(oldCluster.Status.PluginStatus, cluster.Status.PluginStatus) {
+		return nil
+	}
+
 	return r.Client.Status().Patch(ctx, cluster, client.MergeFrom(oldCluster))
```

Comment:
Shouldn't this already avoid hitting the api server if the patch is empty (which should always be the case if deepequal)?

Reply:
Nice question. Turns out it doesn't :(

Reply:
@sxd have more details

Reply:
@phisco I did explain you this or not? because I don't remember :S

Reply:
Yes you did, the issue wasn't hitting the api server, which it already did for no-op changes, but all the json marshaling/unmarshaling to build patches and understand it was a no-op. So the pr description could be updated, but the fix is right.